src/wp-admin/includes/media.php

```diff
@@ -2204,6 +2204,11 @@ function media_upload_form( $errors = null ) {
 		$plupload_init['avif_upload_error'] = true;
 	}
 
+	// Check if HEIC images can be edited.
+	if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/heic' ) ) ) {
+		$plupload_init['heic_upload_error'] = true;
```

Comment:
`heic_upload_error` is already set in `wp_show_heic_upload_error` which is hooked into the `plupload_init` filter.

This seems the more logical place, though, so maybe we can deprecate `wp_show_heic_upload_error`.

Reply:
good point, I was just trying to copy all the places it was added for avif, trying to exclude those having to do with display (we aren't trying to display HEIC, only convert uploads to JPEG).

Reply:
Oh, I see now. Let me check for any plugin use of `wp_show_heic_upload_error` and consider deprecating it. Otherwise, I'll at least add the mime type supports check there.

Reply:
nothing that would break plugins, however I see that when this filter was added, @azaozz commented that is specifically added via filters so that it can be removed. for that reason, I'm going to move this code to the filter and remove the use here.

Reply:
Fixed in 683a076401d221a65ee2657e92897b00086314cc